rmw_iceoryx_cpp/src/rmw_serialize.cpp

```diff
@@ -28,8 +55,37 @@ rmw_serialize(
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(type_supports, RMW_RET_ERROR);
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(serialized_message, RMW_RET_ERROR);
 
-  assert(false);
-  return RMW_RET_OK;
+  // it's a fixed size message, so we memcpy
+  if (rmw_iceoryx_cpp::iceoryx_is_fixed_size(type_supports)) {
+    return details::copy_payload_into_serialized_message(
+      serialized_message, ros_message, rmw_iceoryx_cpp::iceoryx_get_message_size(type_supports));
+  }
+
+  // message is neither loaned nor fixed size, so we have to serialize
```

Comment:
I guess this comment is a copy paste from rmw_publish(). We have no loan handling here, so maybe say something like '// message is not fixed size, so we have to serialize'

Reply:
done